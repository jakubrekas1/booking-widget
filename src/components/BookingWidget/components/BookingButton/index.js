export { default as BookingButton } from './BookingButton';
