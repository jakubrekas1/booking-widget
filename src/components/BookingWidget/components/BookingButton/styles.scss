@import '../../../../assets/styles/colors';

button::-moz-focus-inner {
  border: 0;
}

.booking-button{
  display: block;
  width: 100%;
  padding: 12px 30px;
  background-color: $op;
  border-width: 0;
  border-radius: 4px;
  color: $white;
  font-size: 24px;
  cursor: pointer;
  transition: 100ms all linear;

  &:hover,
  &:focus{
    outline: none;
    background-color: darken($op, 5%);
  }

  &:disabled {
    opacity: 0.4;
    cursor: not-allowed;
  }
}
