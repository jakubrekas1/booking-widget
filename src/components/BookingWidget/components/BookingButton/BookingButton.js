import React from 'react';
import PropTypes from 'prop-types';

import './styles.scss';

const BookingButton = ({ disabled, onClick }) => (
  <button
    className="booking-button"
    disabled={disabled}
    onClick={onClick}
    type="submit"
  >
    Book now
  </button>
);

BookingButton.defaultProps = {
  disabled: false,
};

BookingButton.propTypes = {
  disabled: PropTypes.bool,
  onClick: PropTypes.func.isRequired,
};

export default BookingButton;
