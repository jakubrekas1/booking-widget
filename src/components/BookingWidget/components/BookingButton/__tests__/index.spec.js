import React from 'react';
import renderer from 'react-test-renderer';
import BookingButton from '../BookingButton';

describe('Booking button component', () => {
  const Component = <BookingButton onClick={() => null} />;
  it('should renders correctly', () => {
    const tree = renderer.create(Component).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
