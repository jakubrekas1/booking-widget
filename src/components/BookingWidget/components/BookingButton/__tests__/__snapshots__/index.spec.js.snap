// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Booking button component should renders correctly 1`] = `
<button
  className="booking-button"
  onClick={[Function]}
  type="submit"
>
  Book now
</button>
`;
